### DISCUSSION TITLE
Why should variable overrides must come after the functions are imported, while customizing bootstrap?

### DISCUSSION THREAD

**Discussion by AnupamKhosla** [twbs/bootstrap#36602] [Q&A] [answered]
The third paragraph in [Variables Default](https://getbootstrap.com/docs/5.2/customize/sass/#variable-defaults) subsection of `Customize > Sass` page, mentions:  
> Variable overrides must come after our functions are imported, but before the rest of the imports.  

Why is that? Don't we want the `_functions.scss` to use our project's custom variables instead of bootstrap's default variables?

**Comment by SamuelMiller** [comment]
I too am curious, particularly since just including @import '../node_modules/bootstrap/scss/bootstrap'; after declaring custom variables works.

**Comment by mdo** [comment, marked as answer]
This is covered a bit in our docs: https://getbootstrap.com/docs/5.2/customize/sass/#importing. If you include the whole of Bootstrap, you can place your variables first, but you may have issues with variable and map re-use later in our codebase. In addition, functions for things like SVGs and colors and more won't be available.

In contrast to importing individual pieces, you get full control and access to the functions. There are a few code comments that talk about this in the Customize> Sass page.

> // Include any default variable overrides here (though functions won't be available)

> 1. Include functions first (so you can manipulate colors, SVGs, calc, etc)
